From c26d8b0554f162ca2b05f0960b4382956f561c48 Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Thu, 31 Jul 2014 17:23:17 +0800
Subject: [PATCH 519/635] ENGR00324895 CTS-Verifier/Camera Formats: buffer
 size of all YUY2

formats were mismatched. 100%

Due to alignment, gralloc may allocate size larger than actual size.
So use actual size as preview frame size.

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 mx6/libcamera/CameraBridge.cpp | 21 ++++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/mx6/libcamera/CameraBridge.cpp b/mx6/libcamera/CameraBridge.cpp
index a754fd6..196c8d5 100644
--- a/mx6/libcamera/CameraBridge.cpp
+++ b/mx6/libcamera/CameraBridge.cpp
@@ -434,7 +434,26 @@ status_t CameraBridge::start()
         return NO_INIT;
     }
 
-    mPreviewBufferSize = mFrameProvider->getFrameSize();
+    int previewWidth = 0;
+    int previewHeight = 0;
+
+    mParameters.getPreviewSize(&previewWidth, &previewHeight);
+    const char *previewFormat = mParameters.getPreviewFormat();
+
+    if(previewFormat) {
+        if ( strcmp(previewFormat, "yuv422i-yuyv") == 0 ) {
+            mPreviewBufferSize = previewWidth * previewHeight * 2;
+        } else if ( (strcmp(previewFormat, "yuv420p") == 0) || (strcmp(previewFormat, "yuv420sp") == 0) ) {
+            mPreviewBufferSize = previewWidth * previewHeight * 3/2;
+        } else {
+            ALOGE("CameraBridge::start, not support format %s", previewFormat);
+            mPreviewBufferSize = previewWidth * previewHeight * 3/2;
+        }
+    } else {
+        ALOGE("CameraBridge::start, can't get preview format");
+            mPreviewBufferSize = previewWidth * previewHeight * 3/2;
+    }
+
 
 #ifdef EVK_6SL //driver provide yuyv, but h264enc need nv12
     int bufSize = mFrameProvider->getFrameSize() * 3/4;
-- 
1.8.0

